Adyen/Utilities/PublicKeyProvider/PublicKeyConsumer.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// :nodoc:
+public protocol PublicKeyConsumer: PaymentComponent {
+
+    /// Provider for fetching the public key.
+    var publicKeyProvider: AnyPublicKeyProvider { get set }
+    
+    /// Convenient way to fetch the client public key with a closure for the success case
+    /// and an option to notify the delegate on the failure case.
+    /// - Parameters:
+    ///   - notifyingDelegateOnFailure: If `true`, notifies the `PaymentComponentDelegate` on failure.
+    ///   - successHandler: The block that is called when fetching was successful. Contains the public key.
+    func fetchCardPublicKey(notifyingDelegateOnFailure: Bool, successHandler: PublicKeySuccessHandler?)
```

Comment:
this is a bit of an unusual pattern. can we change this to pass an optional completion `_ result: Result<String, Error>` here? the way this is used right now, it is either passing `true` and a `successHandler` (so the consumer cares about both outcomes) or passing `false` and no handler (consumer does not care about the outcome). this would also simplify the `fetchCardPublicKey` implementation in the extension.
i understand that `PublicKeyConsumer` extends `PaymentComponent`, that's why this pattern was implemented. i wonder if it makes sense to split it from `PaymentComponent` and change the `fetchCardPublicKey`

Reply:
Had the same conversation with Mohamed. Wanted to have a better way since we are making it public now, but ended up this way as if we go with the result, it will be same signature as the one this calls inside.